### PR TITLE
dockerise

### PR DESCRIPTION
**Pull request by sandlerr**
@jonmoter I was working on this last week, may be of some use

### CODE REVIEW

**Comment by sandlerr** [comment]
Something is up with mysql on travis, they're rolling back some changes

File: config/initializers/database.rb [hunk 1]
```diff
@@ -3,7 +3,10 @@
 
   ActiveSupport.on_load(:active_record) do
     config = Rails.application.config.database_configuration[Rails.env]
-    config['pool'] = (ENV['DB_POOL'] || 100).to_i
-    ActiveRecord::Base.establish_connection(config)
+
+    if config
+      config['pool'] = (ENV['DB_POOL'] || 100).to_i
+      ActiveRecord::Base.establish_connection(config)
+    end
```
Comment (jonmoter) [inline config/initializers/database.rb]: What are the scenarios where we'd want to run the database initializer but not attempt to connect to the database?

Comment (sandlerr) [inline config/initializers/database.rb]: to precompile assets afaik. This is Steve D's work.

**Comment by sandlerr** [comment]
cc @zendesk/samson

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+FROM ruby:2.2.2
+
+RUN apt-get update
+RUN apt-get install -y wget apt-transport-https
```
Comment (grosser) [inline Dockerfile]: these should be 1 line or you end up in a place where the update is cached and the install fails

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Code
+ADD config /app/config
+ADD app /app/app
+ADD lib /app/lib
+
+RUN DATABASE_URL=mysql2://user:pass@127.0.0.1/null RAILS_ENV=development PRECOMPILE=1 bundle exec rake --trace assets:precompile
+
+EXPOSE 3000
+
+CMD DATABASE_URL=$MYSQL_URL bundle exec rails server --binding=0.0.0.0
```
Comment (grosser) [inline Dockerfile]: should this be puma ?

**Comment by grosser** [comment]
:+1:

File: Dockerfile (hunk 3 above)
Comment (sandlerr) [inline Dockerfile]: `rails s` magically boots puma. We could make it explicit (and use `config/puma.rb`) but I'm not sure if that matters for development

Comment (grosser) [inline Dockerfile]: this is meant for production too
I don't think rails s actually loads puma.rb ... ran into this issue on
another project ...

Comment (sandlerr) [inline Dockerfile]: sure, will fix

**Pull request merged**